actionpack/test/controller/test_case_test.rb

```diff
@@ -262,11 +262,11 @@ def to_param
   end
 
   def test_body_stream
-    params = Hash[:page, { name: "page name" }, "some key", 123]
+    body = "Hello World"
 
-    post :render_body, params: params.dup
+    post :render_body, body: body
```

Comment:
I don't think setting the body directly is the right fix here. 

This would [bypass the `body_stream` branch](https://github.com/rails/rails/blob/main/actionpack/lib/action_dispatch/http/request.rb#L356-L362) in `ActionDispatch::Request#body` and [there is already have a test for it.](https://github.com/rails/rails/blob/main/actionpack/test/controller/test_case_test.rb#L277-L280)

This is currently failing on CI, so I tried a different fix and submitted a [PR](https://github.com/rails/rails/pull/48158) targeting main to bring back CI to green.